lib/vehicle_factory.rb

```diff
@@ -0,0 +1,38 @@
+require './vehicle'
+
+class VehicleFactory
+  attr_accessor :created_vehicles
+  
+  def initialize
+    @created_vehicles = []
+  end
+
+  def create_vehicles_by_state(state_database)
+    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    ny_state_registrations = DmvDataService.new.ny_dmv_office_locations
+
+    if state_database == wa_ev_registrations
+      wa_ev_registrations.each do |car|
+        @created_vehicles << Vehicle.new({
+            vin: car[:vin_1_10],
+            year: car[:model_year],
+            make: car[:make],
+            model: car[:model],
+            engine: 'ev'
+        })
+      end
+        @created_vehicles
```

Comment:
I would also try to refactor out the formatting of the data so this method can focus only on creating vehicle objects. It might look something like this:
```ruby
def create_vehicles_by_state(state_database)
    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
    ny_state_registrations = DmvDataService.new.ny_dmv_office_locations

    if state_database == wa_ev_registrations
      wa_registrations = format_wa_data(state_database)
      wa_registrations.each do |car_data|
        @created_vehicles << Vehicle.new(car_data)
      end
    elsif state_database == ny_state_registrations
      vehicle_data.map do |vehicle|
        @created_vehicles << Vehicle.new({
            vin: vehicle[:vin],
            year: vehicle[:model_year],
            make: vehicle[:make],
            model: nil,
            engine: 'ev'
          })
      end
    end
    @created_vehicles
  end

  def format_wa_data(wa_ev_registrations)
    formatted_car_info = {}
    wa_ev_registrations.each do |car|
      formatted_car_info[:vin] = car[:vin_1_10],
      formatted_car_info[:year] = car[:model_year],
      formatted_car_info[:make] = car[:make],
      formatted_car_info[:model] = car[:model],
      formatted_car_info[:engine] = 'ev'
    end
  end
```
I would do the same for New York too but I'll leave that for you to play with.